src/main/java/de/konfidas/ttc/messages/LogMessage.java

```diff
@@ -120,6 +121,36 @@ public String getSignatureAlgorithm(){
 
     public BigInteger getSignatureCounter(){ return signatureCounter; }
 
+    private byte[] getLFromASN1(ASN1Primitive element) throws IOException{
+        byte[] elementContent = element.getEncoded();
+
+        if (((int)elementContent[0])<=128){
+            //Case: Definite length encocding, one byte oder indefinte endcding
+            return Arrays.copyOfRange(elementContent, 1, 2);
+        }
+        else {
+            //Case: Definite length encocding, multple bytes
+            int elementNumberOfLengthBytes = (int)elementContent[0]-128;
+             return Arrays.copyOfRange(elementContent, 1, elementNumberOfLengthBytes+1);
+        }
+
+    }
+
+    byte[] getVFromASN1(ASN1Primitive element) throws IOException{
+        byte[] elementContent = element.getEncoded();
+
+        if (((int)elementContent[0])<=128){
+            //Case: Definite length encocding, one byte oder indefinte endcding
+            return Arrays.copyOfRange(elementContent, 2, elementContent.length);
+        }
+        else {
+            //Case: Definite length encocding, multple bytes
+            int elementNumberOfLengthBytes = (int)elementContent[0]-128;
+             return Arrays.copyOfRange(elementContent, 2 + elementNumberOfLengthBytes, elementContent.length);
```

Comment:
use the getLength Method above, then `BigInteger.valueof(1, getLength()))` to copy only the correct length.

On the other hand, if elementContent.length gives you the right length, why do you not use it estimate the encoding size? I think of something like `Math.ceil(BigInteger.valueOf(elementContent.length). bitCount​() / 8) `but this in itself already looks quite ugly.